making-simfiles-mods-attacks.md

```diff
@@ -0,0 +1,160 @@
+
+(The default `*1` applies at a rate of 100% of a mod over 1 second)
+
+
+> Mod apply speed works for **any strength**, including **0%**! If you want to get rid of a mod really fast, you can use an attack like `*50 0% Beat`.
+
+![The difference between *0.5 Beat and *10 Beat](/_media/making-simfiles/mods-attacks-beat-speed.m4v ':include :type=video')
+
+> In the above video - **Left = \*0.5 Beat** (fades in half as fast), **Right = \*10 Beat** (fades in 10x as fast!). They both get to the **same strength** at the end, but the **left side takes longer** to do so.
+
+## Resources/What's Next
+
+Next up is to play around with Modifiers! Experiment with what effects you can get, and discover what you do/don't like!
+
+---
+
+### Mods Resources
+
+In the future, we'll better document what mods you can use on this wiki. For now:
+
+ * In Outfox, go to the **player options** menu after picking a song (press the start button twice). Scroll to the bottom, pick **Modifiers** and then **Exit**. Most Modifiers in this list (namely in the **Acceleration, Effects, Appearance, and Scroll** categories) work well as Mods in Attacks!
+ 
+ * **[FMS Cat's Flip-Invert site](http://fms-cat.com/flip-invert/)** helps give you combinations of various mods to move the note columns across the screen. Just drag the receptors where you want them on-screen!
+    * (You can alternatively use the `MoveX/Y/Zn` mods to move single note columns, like `MoveX1`, `MoveX2`, etc)
+ 
+ * Check out **[WinDEU's Marathon Tutorial](https://www.youtube.com/watch?v=4cHCqwHcKPY&list=PL1joaptC-JIUphz_wcd8WrMCdeSylZmdw&index=3)** series - it focuses on older versions of StepMania (where some mods, like Drunk and Tipsy, have different names), but a lot of the info still applies to SM5/Outfox. The later videos in the series dive *really* deep into how some mods behave!
+    * How to set up Mods in Courses in also covered in earlier videos in the series (though again, focusing on older versions of StepMania)
+ 
+ * You can join **[Project Outfox's Discord server](https://discord.gg/cN4TjgQdcA)** for more help from the community - feel free to ask mods/attacks-related questions in channels like `#mods-coding-help`.
+
+---
+
+### My Mods Don't Work!
+
+  * Double check you've spelled your Mod name right - mods from older versions of the game (like OpenITG) may have different names in SM5/Outfox!
+  * You may get strange behavior if you add new attacks into a simfile before the last attack in the file.
+
+---
+
+### Mod Strengths Across Different Themes
+
+The mods you add will change strength depending on the internal resolution of whatever theme you're currently using.
+
+If you switch between an HD theme (like the Outfox or SM5.1 default themes) and a non-HD theme (like SM5.0.12's default theme or PIU Infinity), you'll find **Mods will be stronger** on **non-HD themes**.
+
+Plan accordingly if you plan on playing your simfiles/chart edits on different themes!
```

Comment:
So, this hasn't been true for a while now in OutFox (likely before the first public alpha, even). Modifiers like Drunk used to use SCREEN_HEIGHT in their calculations, but now they use a metric called "ModifierHeight". If you're making this for both OutFox and 5.x, be sure to specify where people which modifiers (and which version) have this oddity.

Reply:
our wiki should just be our version to prevent confusion, i'd say we can mention how it was in 5.x, but we should have this targeted for our version.

Reply:
Noted, and noted - ModifierHeight is new to me, and very good to know about - I see it in the metrics for the fallback theme now that you mention it.

I'll go ahead and put in a mention that it exists (and how it differs from 5.x,). If there ends up being a page in the future that lists the effects of all the mods [(a la mainline SM's wiki)](https://github.com/stepmania/stepmania/wiki/List-of-Song-Modifiers), I figure that would be a good place to mention specific mods ModifierHeight applies to (aside from a few common examples, here).

*(Sidenote: While I'm at it, is there a more proper term for the option to make mods take longer to take effect than "Application Speed"? I kinda took an educated guess, and if there's more official terminology I'd like to use it.)*

Reply:
re: "Application Speed", I've called it "Approach Rate", but I don't know if that set of words makes sense to everyone.